### PR TITLE
New design: package list page.

### PR DESCRIPTION
**Pull request by isoos**
This has a lot of overlap with the `/search` page, with the notable differences being:
- search has no platform-specific title and description
- package list page has no count on the number of packages
- package list includes the author (in the old design), while the search page didn't (in both) - TBD later

At first I've tried to merge this with the search page in one go, but ended up not doing that, rather doing it in separate steps (too many knobs involved, which may be better to do in isolation). In the end I expect that we'll have a unified `/packages` page with both the search functions and the platform switcher.

### CODE REVIEW

**Comment by mkustermann** [comment]
If you update this PR, I'll review and we can land it & do a prod deployment with `/experimental` :)

File: app/lib/frontend/templates.dart [hunk 1]
```diff
@@ -124,6 +124,66 @@ class TemplateService {
...
+
+    final platformTabs = [
+      platformTabData('Flutter', KnownPlatforms.flutter),
+      platformTabData('Web', KnownPlatforms.web),
+      platformTabData('Server', KnownPlatforms.server),
+      platformTabData('All', null),
+    ];
+
+    final values = {
+      'platform_tabs': platformTabs, // experimental_design: new
```
Comment (mkustermann) [inline app/lib/frontend/templates.dart]: I don't even understand what `// experimental_design: new` means. Maybe just get rid of these comments. If the map entries are used in old/new templates, maybe rephrase these to `// used in v2 only`.

Comment (isoos) [reply app/lib/frontend/templates.dart]: yeah, I'll do a followup to change everything. it was meant to be a unique string that is easy to search for when we'll do the merging and cleanup, a shorter version works just as well. (initially I was aiming for naming everything as "experimental", but that is no longer the case)
